### PR TITLE
[TACHYON-865] use TachyonConf for initialization in block client

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-865

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java [hunk 1]
```diff
@@ -45,9 +46,9 @@
    * @param workerAddress the worker address
    */
   public BlockWorkerClientPool(NetAddress workerAddress) {
-    // TODO(calvin): Get the capacity from configuration.
-    super(CAPACITY);
-    mExecutorService = Executors.newFixedThreadPool(CAPACITY, ThreadFactoryUtils.build(
+    super(ClientContext.getConf().getInt(Constants.USER_BLOCK_WORKER_CLIENT_THREADS));
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java]: actually  this is probably better to be `USER_LOCAL_BLOCK_WORKER_CLIENT_THREADS` with string `tachyon.user.local.block.worker.client.threads`?

File: clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java [hunk 2]
```diff
@@ -53,8 +53,7 @@
    */
   BlockStoreContext() {
     mBlockMasterClientPool = new BlockMasterClientPool(ClientContext.getMasterAddress());
-    // TODO(calvin): Get the capacity from configuration.
-    final int CAPACITY = 10;
+    final int CAPACITY = ClientContext.getConf().getInt(Constants.USER_REMOTE_WORKER_THREADS);
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java]: maybe `USER_REMOTE_BLOCK_WORKER_THREADS`

File: clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java [hunk 3]
```diff
@@ -40,10 +41,10 @@
    * @param masterAddress the master address
    */
   public BlockMasterClientPool(InetSocketAddress masterAddress) {
-    // TODO(calvin): Get the capacity from configuration.
-    super(CAPACITY);
+    super(ClientContext.getConf().getInt(Constants.USER_BLOCK_MASTER_CLIENT_THREADS));
+    int capacity = ClientContext.getConf().getInt(Constants.USER_BLOCK_MASTER_CLIENT_THREADS);
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java]: Oh I meant you can use `mMaxCapacity`.

File: clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java (hunk 1 above)
Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockWorkerClientPool.java]: done

File: clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java (hunk 2 above)
Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java]: done

File: clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java (hunk 3 above)
Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java]: done

File: clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java [hunk 4]
```diff
@@ -30,9 +32,9 @@
  * using the client.
  */
 public class BlockMasterClientPool extends ResourcePool<BlockMasterClient> {
-  private static final int CAPACITY = 10;
   private final ExecutorService mExecutorService;
   private final InetSocketAddress mMasterAddress;
+  private final int mMaxCapacity;
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java]: Isn't `mMaxCapacity` already defined in `ResourcePool`?

File: clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java [hunk 5]
```diff
@@ -53,8 +53,7 @@
    */
   BlockStoreContext() {
     mBlockMasterClientPool = new BlockMasterClientPool(ClientContext.getMasterAddress());
-    // TODO(calvin): Get the capacity from configuration.
-    final int CAPACITY = 10;
+    final int CAPACITY = ClientContext.getConf().getInt(Constants.USER_REMOTE_BLOCK_WORKER_THREADS);
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java]: since this is just a local variable, just use `int capacity`?

Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockStoreContext.java]: done

File: clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java [hunk 6]
```diff
@@ -40,10 +41,8 @@
    * @param masterAddress the master address
    */
   public BlockMasterClientPool(InetSocketAddress masterAddress) {
-    // TODO(calvin): Get the capacity from configuration.
-    super(CAPACITY);
-    mExecutorService =
-        Executors.newFixedThreadPool(CAPACITY,
+    super(ClientContext.getConf().getInt(Constants.USER_BLOCK_MASTER_CLIENT_THREADS));
+    mExecutorService = Executors.newFixedThreadPool(mMaxCapacity,
             ThreadFactoryUtils.build("block-master-heartbeat-%d", true));
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/block/BlockMasterClientPool.java]: is this the right amount of indenting?

File: common/src/main/resources/tachyon-default.properties [hunk 7]
```diff
@@ -103,3 +103,6 @@ tachyon.user.localread.enable=true
 tachyon.user.localwrite.enable=true
 tachyon.user.network.netty.worker.threads=0
 tachyon.user.network.netty.timeout.ms=3000
+tachyon.user.remote.block.worker.threads=10
```
Comment (gpang) [inline common/src/main/resources/tachyon-default.properties]: Can you add comments on what exactly these parameters are controlling?

**Comment by gpang** [comment]
I left a few questions. Thanks for the PR!

File: common/src/main/resources/tachyon-default.properties (hunk 7 above)
Comment (miasantreble) [inline common/src/main/resources/tachyon-default.properties]: done

Comment (apc999) [inline common/src/main/resources/tachyon-default.properties]: the local one is called `tachyon.user.local.block.worker.client.threads`
but the remote one is called `tachyon.user.remote.block.worker.threads` without `client`,
is it by purpose?

Comment (miasantreble) [inline common/src/main/resources/tachyon-default.properties]: so call it `tachyon.user.remote.block.worker.client.threads` and update the docs to something like `"How many threads to use for remote block worker client"`?

File: docs/Configuration-Settings.md [hunk 8]
```diff
@@ -448,6 +448,21 @@ The user configuration specifies values regarding file system access.
   <td>3000</td>
   <td>The maximum number of milliseconds for a netty client (for block reads and block writes) to wait for a response from the data server.</td>
 </tr>
+<tr>
+  <td>tachyon.user.remote.block.worker.threads</td>
+  <td>10</td>
+  <td>How many threads to use for remote block worker.</td>
```
Comment (apc999) [inline docs/Configuration-Settings.md]: it is still the number threads in the pool waiting for clients talking to a (arbitrary) remote block worker. So maybe name the property name consistent ?

**Comment by apc999** [comment]
Left two minor comments. Thanks!

**Comment by apc999** [comment]
+1 LGTM. Thanks for the patch!

**Comment by calvinjia** [comment]
@miasantreble Looks good overall, could you resolve the merge conflicts?

File: docs/Configuration-Settings.md [hunk 9]
```diff
@@ -465,6 +465,26 @@ The user configuration specifies values regarding file system access.
...
 </tr>
+<tr>
+  <td>tachyon.user.remote.block.worker.client.threads</td>
+  <td>10</td>
+  <td>How many threads to use for remote block worker client.</td>
+</tr>
+<tr>
+  <td>tachyon.user.local.block.worker.client.threads</td>
+  <td>10000</td>
+  <td>How many threads to use for block worker client pool.</td>
```
Comment (apc999) [inline docs/Configuration-Settings.md]: ... to read from a local block worker.

File: docs/Configuration-Settings.md [hunk 10]
```diff
@@ -465,6 +465,26 @@ The user configuration specifies values regarding file system access.
   <td>3000</td>
   <td>The maximum number of milliseconds for a netty client (for block reads and block writes) to wait for a response from the data server.</td>
 </tr>
+<tr>
+  <td>tachyon.user.remote.block.worker.client.threads</td>
+  <td>10</td>
+  <td>How many threads to use for remote block worker client.</td>
```
Comment (apc999) [inline docs/Configuration-Settings.md]: ... to read from a remote block worker.

File: docs/Configuration-Settings.md [hunk 11]
```diff
@@ -465,6 +465,26 @@ The user configuration specifies values regarding file system access.
...
+</tr>
+<tr>
+  <td>tachyon.user.local.block.worker.client.threads</td>
+  <td>10000</td>
+  <td>How many threads to use for block worker client pool.</td>
+</tr>
+<tr>
+  <td>tachyon.user.block.master.client.threads</td>
+  <td>10</td>
+  <td>How many threads to use for block master client pool.</td>
```
Comment (apc999) [inline docs/Configuration-Settings.md]: ... for block master client to talk to block master

**Comment by miasantreble** [comment]
@apc999 docs updated, thanks!

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
+1 LGTM

**Comment by apc999** [comment]
@calvinjia , can you take a look? Thanks.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**